src/pages/workspaces/workspace/Data.test.ts

```diff
@@ -0,0 +1,137 @@
+
+describe('WorkspaceData', () => {
+  // When Data.js is broken apart and the WorkspaceData component is converted to TypeScript,
+  // this type belongs there.
+  interface WorkspaceDataProps {
+    namespace: string;
+    name: string;
+    workspace: WorkspaceWrapper;
+    refreshWorkspace: () => void;
+    storageDetails: StorageDetails;
+  }
+
+  // SIFERS setup, see: https://medium.com/@kolodny/testing-with-sifers-c9d6bb5b362
+  function setup({
+    namespace = 'test-namespace',
+    name = 'test-name',
+    workspace,
+    refreshWorkspace = () => {},
+    storageDetails = {},
+    status = 'RUNNING' as LeoAppStatus,
+  }) {
+    type AjaxContract = ReturnType<typeof Ajax>;
+    type WorkspacesContract = AjaxContract['Workspaces'];
+    type AppsContract = AjaxContract['Apps'];
+    type WorkspaceDataContract = AjaxContract['WorkspaceData'];
+
+    const wdsApp: DeepPartial<ListAppResponse> = {
+      proxyUrls: {
+        wds: 'https://fake.wds.url/',
+      },
+      appType: 'WDS',
+    };
+
+    const mockGetSchema = jest.fn().mockResolvedValue([]);
+    const mockAjax: DeepPartial<AjaxContract> = {
+      Workspaces: {
+        workspace: (_namespace, _name) => ({
+          details: jest.fn().mockResolvedValue(workspace),
+          listSnapshots: jest.fn().mockResolvedValue([]),
+          entityMetadata: jest.fn().mockResolvedValue({}),
+        }),
+      } as DeepPartial<WorkspacesContract>,
+      WorkspaceData: {
+        getSchema: mockGetSchema,
+      } as DeepPartial<WorkspaceDataContract>,
+      Apps: {
+        listAppsV2: jest.fn().mockResolvedValue([{ ...wdsApp, status }]),
+      } as DeepPartial<AppsContract>,
+    };
+
+    asMockedFn(Ajax).mockImplementation(() => mockAjax as AjaxContract);
+
+    const workspaceDataProps = {
+      namespace,
+      name,
+      workspace,
+      refreshWorkspace,
+      storageDetails,
+    } as WorkspaceDataProps;
+
+    return { workspaceDataProps, mockGetSchema };
+  }
+
+  it('displays a waiting message for an azure workspace that is still provisioning in WDS', async () => {
+    // Arrange
+    const { workspaceDataProps } = setup({ workspace: defaultAzureWorkspace, status: 'PROVISIONING' });
+
+    // Act
+    await act(() => {
+      render(h(WorkspaceData, workspaceDataProps));
+    });
+
+    // Assert
+    expect(screen.getByText(/Preparing your data tables/)).toBeVisible();
+    expect(screen.queryByText(/Data tables are unavailable/)).toBeNull(); // no error message
+  });
+
+  it('displays an error message for an azure workspace that fails when loading schema info', async () => {
+    // Arrange
+    const { workspaceDataProps, mockGetSchema } = setup({ workspace: defaultAzureWorkspace, status: 'RUNNING' });
+
+    mockGetSchema.mockRejectedValue(new Error('schema error'));
+
+    // Act
+    await act(() => {
+      render(h(WorkspaceData, workspaceDataProps));
+    });
+
+    // Assert
+    expect(screen.getByText(/Data tables are unavailable/)).toBeVisible();
```

Comment:
```suggestion
    expect(screen.getByText('Data tables are unavailable')).toBeVisible();
```

Style nit: suggest using a string unless a regex is necessary.

Reply:
I landed on regex because using the string I needed to include the entire text of the DOM element to get a hit -- this kinda felt brittle to me.  I didn't want a simple text/grammar change/addition/clarification to cause a _definite_ test failure.    Happy to go full string though if, even with this tradeoff in mind, you'd still recommend string.

Reply:
That makes sense.